File: typegraph/python_next/typegraph_next/wit.py
# Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
# SPDX-License-Identifier: MPL-2.0

from wasmtime import Store

from typegraph_next.gen import Root, RootImports
from typegraph_next.gen.exports.core import Core
from typegraph_next.gen.exports.runtimes import Runtimes
from typegraph_next.gen.exports.utils import Utils
from typegraph_next.gen.exports.aws import Aws
from typegraph_next.imports import Abi

store = Store()
abi = Abi()
_typegraph_core = Root(store, RootImports(abi=abi))

core = Core(_typegraph_core)
runtimes = Runtimes(_typegraph_core)
aws = Aws(_typegraph_core)
wit_utils = Utils(_typegraph_core)

File: src/typegraph/deno/src/wit.ts
// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
// SPDX-License-Identifier: MPL-2.0

import type { MetatypeTypegraphCore } from "./gen/typegraph_core.d.ts";
import type { MetatypeTypegraphRuntimes } from "./gen/typegraph_core.d.ts";
import type { MetatypeTypegraphAws } from "./gen/typegraph_core.d.ts";
import type { MetatypeTypegraphUtils } from "./gen/typegraph_core.d.ts";
import * as js from "./gen/typegraph_core.js";

export const core = js.core as typeof MetatypeTypegraphCore;
export const runtimes = js.runtimes as typeof MetatypeTypegraphRuntimes;
export const aws = js.aws as typeof MetatypeTypegraphAws;
export const wit_utils = js.utils as typeof MetatypeTypegraphUtils;

export type { Cors, Rate } from "./gen/typegraph_core.d.ts";
export type {
  Auth,
  AuthProtocol,
  AuthProtocolBasic,
  AuthProtocolJwt,
  AuthProtocolOauth2,
} from "./gen/typegraph_core.d.ts";
